mutis/signal.py

```diff
@@ -30,37 +27,42 @@ class Signal:
     ----------
     times : :class:`numpy.ndarray` or :class:`pandas.Series`
         Values of the time axis.
-    signs : :class:`numpy.ndarray` or :class:`pandas.Series`
+    values : :class:`numpy.ndarray` or :class:`pandas.Series`
         Values of the signal axis.
-    method : :class:`str`
+    fgen : :class:`str`
         Method used to generate the synthetic signal.
     """
 
-    def __init__(self, times, signs, method):
+    def __init__(self, times, values, fgen):
         self.times = np.array(times)
-        self.signs = np.array(signs)
-        self.method = method
+        self.values = np.array(values)
+        self.fgen = fgen
         self.synth = None
 
         # TODO make attributes below specific of OU method / not the entire class
         self.theta = None
         self.mu = None
         self.sigma = None
 
-    def gen_synth(self, N):
+    def gen_synth(self, samples):
         """Description goes here."""
 
-        self.synth = np.empty((N, self.times.size))
-        for n in range(N):
-            if self.method == "lc_gen_samp":
-                self.synth[n] = lc_gen_samp(self.signs)
-            if self.method == "lc_gen_psd":
-                self.synth[n] = lc_gen_psd(self.signs, self.times)
-            if self.method == "lc_gen_ou":
+        self.synth = np.empty((samples, self.times.size))
+        for n in range(samples):
+            if self.fgen == "lc_gen_samp":
+                self.synth[n] = lc_gen_samp(self.values)
+            elif self.fgen == "lc_gen_psd_nft":
+                self.synth[n] = lc_gen_psd_nft(self.times, self.values)
+            elif self.fgen == "lc_gen_psd_lombscargle":
+                self.synth[n] = lc_gen_psd_lombscargle(self.times, self.values)
+            elif self.fgen == "lc_gen_psd_fft":
+                self.synth[n] = lc_gen_psd_fft(self.values)
+            elif self.fgen == "lc_gen_ou":
                 if self.theta is None or self.mu is None or self.sigma is None:
                     raise Exception("You need to set the parameters for the signal")
-                else:
-                    self.synth[n] = lc_gen_ou(self.theta, self.mu, self.sigma, self.times)
+                self.synth[n] = lc_gen_ou(self.theta, self.mu, self.sigma, self.times)
+            else:
+                raise(f'Unknown fgen method {self.fgen}')
```

Comment:
replace with

```Python
raise Exception(f"Unknown fgen method {self.fgen}")
```

note also the double quotes in the string, since we are following `black` code-style